to.etc.domui/src/main/java/to/etc/domui/server/AppFilter.java

```diff
@@ -220,15 +228,20 @@ public synchronized void init(final FilterConfig config) throws ServletException
 
 			//-- Are we running in development mode?
 			String domUiReload = DeveloperOptions.getString("domui.reload");
-			String autoload = domUiReload != null ? domUiReload : m_config.getString("auto-reload"); 			// Allow override of web.xml values.
+			String autoload = domUiReload != null ? domUiReload : m_config.getString("auto-reload");            // Allow override of web.xml values.
 
 			//these patterns will be only watched not really reloaded. It makes sure the reloader kicks in. Found bundles and MetaData will be reloaded only.
 			String autoloadWatchOnly = m_config.getString("auto-reload-watch-only");
 
-			if(DeveloperOptions.isDeveloperWorkstation() && DeveloperOptions.getBool("domui.developer", true) && autoload != null && autoload.trim().length() > 0)
+			if(DeveloperOptions.isDeveloperWorkstation() && DeveloperOptions.getBool("domui.developer", true) && autoload != null && autoload.trim().length() > 0) {
 				m_contextMaker = new ReloadingContextMaker(m_applicationClassName, m_config, autoload, autoloadWatchOnly);
-			else
+				if(null == m_isProductionMode)
```

Comment:
This is really odd logic. You are inside the filter's initialization so there should be no need to test for null of that field.

Luckily it needs to be replaced here anyway; you should set the new m_testMode field to true at this point- that is all.